### ISSUE TITLE
Is there any way for initializing multiple instance of cropper on single page?

### ISSUE THREAD

**Issue by RajeshMalviya14** [DominicTobias/react-image-crop#235]
I want to add more than one cropper on the same page but id of cropper seems to be the same for both instances.

**Comment by DominicTobias** [comment]
Hi, when you say ID for the cropper, what do you mean?

**Comment by RajeshMalviya14** [comment]
onCropComplete(crop, pixelCrop) {
     this.getCroppedImg(document.getElementById('cover_image'), pixelCrop)
}
getCroppedImg(image, pixelCrop, fileName) {
}

Thank you for quick response.
I'm picking base64 image from id "cover_image" but we can't have same id for all croppers on page.

**Comment by DominicTobias** [comment]
I'm not sure how you set an ID on the image? But regardless if you look at the demo the image reference is taken from `onImageLoaded` which would work per instance - https://codesandbox.io/s/72py4jlll6

```js
onImageLoaded = (image, pixelCrop) => {
    this.imageRef = image;
  };
```

**Comment by RajeshMalviya14** [comment]
Thank you for your response @DominicTobias .now it work.

**Comment by harrisjb** [comment]
@RajeshMalviya14 would you be willing to share code you used for this to work? Looking into this myself now, and attempting to use onImageLoaded in this way is not working, the last instance of the ReactCropper is always the one effected by the onChange. So if I drag the cropper on the first image, changes appear on the final image. Thank you.

**Comment by DominicTobias** [comment]
Hi @harrisjb it's hard to follow what image is being updated but I didn't want to spend too long debugging exactly what's going wrong because I think that it would be greatly simplified if instead of figuring out which image to update in events, you have a component which you reuse multiple times:

```js
images.map(image => <MyCropComponent key={image.guide} {...image} />
```

That way within `MyCropComponent` you don't have to concern yourself with what to update and the code will be a lot simpler.

As for your bug here: http://recordit.co/obnXVkViCr, thanks for that, it's not something you're doing but something I put in in the latest patch version to automatically adjust invalid aspect crops. It makes using aspect crops simpler but I feared this and will need to try and replicate to see where it's going wrong.

**Comment by harrisjb** [comment]
@DominicTobias @RajeshMalviya14 Just to update you, @DominicTobias ' suggestion did work, moving into a separate component allowed me to keep the correct crop for each instance of the ReactCropper. I will update my example above and touch base with Dominic to see if it would be helpful to add this example to the documentation. Thanks guys.

**Comment by DominicTobias** [comment]
@harrisjb good to hear you sorted it, I'll keep it out the doc for now but might add it to the FAQ if it comes up again thanks. As for the maximum depth error that should be fixed now in the latest version.
